h2o-r/h2o-package/R/frame.R

```diff
@@ -4906,28 +4911,155 @@ h2o.target_encode_apply <- function(data, x, y, target_encode_map, holdout_type,
+#' @return Returns an object containing the target encoding mapping for each column in `x`.
+#' @seealso \code{\link{h2o.target_encode_transform}} for applying the target encoding mapping to a frame.
+#' @export
+h2o.target_encode_fit <- function (data, x, y, fold_column=NULL)
+{
+
+    # Handling `x` parameter
+    if (is.numeric(unlist(x))) {
+        x <- lapply(x, function(i) colnames(data)[i])
+    }
+
+    # Handling `y` parameter
+    if (is.numeric(y)) {
+        y <- colnames(data)[y]
+    }
+
+    # Handling `fold_column` parameter
+    if(missing(fold_column)) {
+        fold_column_name <- ""
+    } else {
+        if (is.numeric(fold_column)) {
+            fold_column <- colnames(data)[fold_column]
+        }
+        fold_column_name <- fold_column
+    }
+
+
+    encoding_map <- .eval.driver(.newExprMap("target.encoder.fit", data, .arrayArgumentHelper(x), .quote(y), .quote(fold_column_name)))
+    return(encoding_map)
+}
+
+#'
+#' Transform Frame by Target Encoding Map
+#' This is an API for a new target encoding implemented in JAVA.
+#'
+#' Applies a target encoding map to an H2OFrame object.  Computing target encoding for high cardinality
+#' categorical columns can improve performance of supervised learning models.
+#'
+#' @param data An H2OFrame object with which to apply the target encoding map.
+#' @param x A list containing the names or indices of the variables to encode. Case when item in the list is a list of multiple columns itself is not supported for now.
+#' @param y The name or column index of the response variable in the data. The response variable can be either numeric or binary.
+#' @param target_encode_map An object that is a result of the calling \code{\link{h2o.target_encode_fit}} function.
+#' @param holdout_type The holdout type used. Must be one of: "LeaveOneOut", "KFold", "None".
+#' @param fold_column (Optional) The name or column index of the fold column in the data. Defaults to NULL (no `fold_column`).
+#' @param blended_avg \code{Logical}. (Optional) Whether to perform blended average. Defaults to TRUE
+#' @param inflection_point  Parameter for blending. Used to calculate `lambda`. Parameter determines half of the minimal sample size for which we completely trust the estimate based on the sample in the particular level of categorical variable.
+#' @param smoothing  Parameter for blending. Used to calculate `lambda`. The parameter f controls the rate of transition between the particular level's posterior probability and the prior probability. For smoothing values approaching infinity it becomes a hard threshold between the posterior and the prior probability.
+#' @param noise (Optional) The amount of random noise added to the target encoding.  This helps prevent overfitting. Defaults to 0.01 * range of y.
+#' @param seed (Optional) A random seed used to generate draws from the uniform distribution for random noise. Defaults to -1.
+#' @param is_train_or_valid \code{Logical}
```

Comment:
@meganjkurka originally I wanted to have some heuristics that could be applied only to the data where we can touch target value. And this required some experiments and discussion( a had todos and comments in java code base for that) when we were releasing it. But we didn't have time and it was left commented out. Another side-effect usage of this parameter is assertions.... it verifies that we don't use KFOLD or LOO on the test data. So for now this parameter is being use only for these assertions. It wasn't not the main point of introducing this parameter. And I feel like we should probably should remove these parameter both from R and Python API's.  @michalkurka can we change Python's API?

Reply:
For validation data, would one need to use KFOLD or LOO?  I thought that the user could use holdout_type = None for any data that is not the data used to create the encoding map.

Reply:
@meganjkurka You are right. Let me explain my original reasoning a little bit. I thought that for blending for the cases when we have too few instances and also for LOO case when we try to apply mapping to single-instance category I can do something better than just using prior frequency computed on the training set. I wanted to treat single-instance(double-instance etc) categories as separate cases. The idea is that probably if some categories are represented by the similar number of instances that probably it means something( eg for name categorical column we could see 2-4 instances in a category in most cases). And even now I believe that there is some knowledge there. But I also realised that we can't probably combine different types of frequencies in one encoding....and we should not do this. Another thing that I was probably wrong about that we can use some heuristics based on data we are applying the encoding map to. I know that it is quite well-established fact that we should not...otherwise we will introduce some kind of leakage. But the main source of confusion for me was the idea that the whole Target encoding is probably about leaking of an experience(frequencies of target column) from training dataset into a new encoding(mapping). That is why I was trying to use some heuristics from data(even from validation data). Now I realised, we are not doing a research here. We need to provide a tool with expected behaviour. So I would remove this parameter from API.